text/3485-feature-documentation.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: feature-documentation
+- Start Date: 2023-09-09
+- RFC PR: [rust-lang/rfcs#3485](https://github.com/rust-lang/rfcs/pull/3485)
+- Rust Issue:
+  [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC describes a new key to under `features` in `Cargo.toml` for
+documentation. This will allow Cargo to display this information to the user and
+provide a way for `rustdoc` to eventually render this data (how this is rendered
+is outside the scope of this RFC).
+
+Please see the parent meta RFC for background information: [`feature-metadata`].
+
+# Motivation
+
+[motivation]: #motivation
+
+Cargo features have become extremely widely used, with many crates having at
+least some level of configuration and larger crates winding up with tens of
+gates. Desipte being a first class component of crate structure, they suffer
+from a documentation problem: users need to maintain documentation separate from
+feature definition, typically a manually-created table within API docs.
+
+This RFC proposes adding feature documentation to `Cargo.toml`, which will allow
+for keeping feature definitions and documentation together.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+A new `doc` key will be allowed within a feature's table. This key provides a
```

Comment:
Naming (not finding the past discussion on it)
- `doc` mirrors `#[doc(...)]`
- `description` mirrors `package.description`
- `documentation` mirrors `package.documentation` (but that is a URL)

We also tend to not use abbreviations as much.  For example, for public-private dependencies, we discussed using `pub` vs `public` and went with `public`

Reply:
@epage In my opinion, I think this should be `doc`.

I think we should encourage using this not just for a "description" of a feature but for "documentation" for a feature, and that's supported by the idea of it showing up in rustdoc and allowing markdown.

And I think there's a big difference in length between `doc` and `documentation`, compared to the difference between `pub` and `public`. (I personally would have gone with `pub` for that too, based on widespread usage of `pub` within Rust, but I also think `public` was less obtrusive because it's fewer characters.)

It's hard enough to get people to write documentation; let's not have any extra friction.

Reply:
>  (I personally would have gone with pub for that too, based on widespread usage of pub within Rust, but I also think public was less obtrusive because it's fewer characters.)

Personally, I'd prefer we be consistent and not just focusing on character count.

> It's hard enough to get people to write documentation; let's not have any extra friction.

We've shot well past that by requiring a "long form" just to write documentation, hurting both discoverability, character count, and ease of not typing some of those characters on international keyboards.